src/System.Linq.Expressions/src/System/Dynamic/DynamicObject.cs

```diff
@@ -229,42 +229,62 @@ public override DynamicMetaObject BindGetMember(GetMemberBinder binder)
             {
                 if (IsOverridden(nameof(DynamicObject.TryGetMember)))
                 {
-                    return CallMethodWithResult(nameof(DynamicObject.TryGetMember), binder, s_noArgs, (e) => binder.FallbackGetMember(this, e));
+                    return BindGetMemberOverridden(binder);
```

Comment:
This doesn't actually need to split into a separate method to avoid the closure, it just needs to not close over anything at the outer scope, e.g.
```C#
if (IsOverridden(nameof(DynamicObject.TryGetMember)))
{
    GetMemberBinder localBinder = binder;
    DynamicMethod localThis = this;
    return CallMethodWithResult(nameof(DynamicObject.TryGetMember), binder, s_noargs, e => localBinder.FallbackGetMember(this, e));
}
```